zarrs/tests/data/hierarchy.zarr/a/foo/zarr.json

```diff
@@ -16,12 +16,20 @@
           "separator": "/"
       }
   },
-  "codecs": [{
+  "codecs": [
+    {
+      "name": "bytes",
+      "configuration": {
+          "endian": "little"
+      }
+    },
```

Comment:
This was was previously invalid since we didn't have an array_to_bytes codec, which means we couldn't write the tests below.